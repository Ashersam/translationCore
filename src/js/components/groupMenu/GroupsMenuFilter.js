import React from 'react';
import PropTypes from 'prop-types';
import { Glyphicon } from 'react-bootstrap';
import InvalidatedIcon from '../svgIcons/InvalidatedIcon';

class GroupsMenuFilter extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      showInvalidated: true,
      showBookmarks: true,
      showSelections: true,
      showNoSelections: true,
      showVerseEdits: true,
      showComments: true
    };
  }

  handleCheckboxSelection(event) {
    const target = event.target;
    const value = target.checked;
    const name = target.name;
    this.setState({
      [name]: value
    });
    console.log(this.state);
    console.log(event);
  }

  render() {
    const styles = {
      color: "var(--reverse-color)",
      width: this.props.menuFilterWidth
    };

    return (
      <div id="groups-menu-filter" style={styles}>
        <div id="groups-menu-filter-top">
          <span id="groups-menu-title">Menu</span><Glyphicon key="filter" glyph="filter"/>
        </div>
        <div id="groups-menu-filter-options">
          <div className="option">
            <span className="object-checkbox"><input type="checkbox" name="showInvalidated" checked={this.state.showInvalidated} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="object-icon"><InvalidatedIcon width={16} height={16} color="#fff"/></span>
            <span className="object-text">Invalidated</span>
          </div>
          <div className="option">
            <span className="option-checkbox"><input type="checkbox" name="showBookmarks" checked={this.state.showBookmarks} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="option-icon"><Glyphicon glyph="bookmark"/></span>
            <span className="option-text">Bookmarks</span>
          </div>
          <div className="option">
            <span className="option-checkbox"><input type="checkbox" name="showSelections" checked={this.state.showSelections} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="option-icon"><Glyphicon glyph="ok"/></span>
            <span className="option-text">Selected</span>
          </div>
          <div className="option">
            <span className="option-checkbox"><input type="checkbox" name="showNoSelections" checked={this.state.showNoSelections} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="option-checkbox"><Glyphicon glyph="ban-circle"/></span>
            <span className="option-text">No Selection</span>
          </div>
          <div className="option">
            <span className="option-checkbox"><input type="checkbox" name="showVerseEdits" checked={this.state.showVerseEdits} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="option-checkbox"><Glyphicon glyph="pencil"/></span>
            <span className="option-text">Verse edit</span>
          </div>
          <div className="option">
            <span className="option-checkbox"><input type="checkbox" name="showComments" checked={this.state.showComments} onChange={this.handleCheckboxSelection.bind(this)} /></span>
            <span className="option-checkbox"><Glyphicon glyph="comment"/></span>
            <span className="option-text">Comments</span>
          </div>
        </div>
      </div>
    );
  }
}

GroupsMenuFilter.defaultProps = {
  menuFilterWidth: '100%'
}

GroupsMenuFilter.propTypes = {
  currentToolName: PropTypes.string.isRequired,
  translate: PropTypes.func.isRequired,
  menuFilterWidth: PropTypes.any
};

export default GroupsMenuFilter;
