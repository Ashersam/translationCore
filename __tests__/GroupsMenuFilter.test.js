/* eslint-disable no-console */
/* eslint-env jest */
import React from 'react';
import GroupsMenuFilter from '../src/js/components/groupMenu/GroupsMenuFilter';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tests for GroupsMenuFilter', () => {
  test('Test GroupsMenuFilter for wordAlignment', () => {
    const props = {
      currentToolName: 'wordAlignment',
      translate: k=>k,
      menuFilterWidth: 238
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilter for translationWords', () => {
    const props = {
      currentToolName: 'translationWords',
      translate: k=>k,
      menuFilterWidth: '100%'
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilter {...props} />
    );

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
