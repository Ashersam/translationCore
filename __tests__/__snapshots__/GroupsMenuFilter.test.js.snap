// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for translationWords 1`] = `
<div
  id="groups-menu-filter"
  style={
    Object {
      "color": "var(--reverse-color)",
      "width": "100%",
    }
  }
>
  <div
    id="groups-menu-filter-top"
  >
    <span
      id="groups-menu-title"
    >
      Menu
    </span>
    <Glyphicon
      bsClass="glyphicon"
      glyph="filter"
      key="filter"
    />
  </div>
  <div
    id="groups-menu-filter-options"
  >
    <div
      className="option"
    >
      <span
        className="object-checkbox"
      >
        <input
          checked={true}
          name="showInvalidated"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="object-icon"
      >
        <InvalidatedIcon
          color="#fff"
          height={16}
          width={16}
        />
      </span>
      <span
        className="object-text"
      >
        Invalidated
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showBookmarks"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-icon"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="bookmark"
        />
      </span>
      <span
        className="option-text"
      >
        Bookmarks
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showSelections"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-icon"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="ok"
        />
      </span>
      <span
        className="option-text"
      >
        Selected
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showNoSelections"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="ban-circle"
        />
      </span>
      <span
        className="option-text"
      >
        No Selection
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showVerseEdits"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="pencil"
        />
      </span>
      <span
        className="option-text"
      >
        Verse edit
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showComments"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="comment"
        />
      </span>
      <span
        className="option-text"
      >
        Comments
      </span>
    </div>
  </div>
</div>
`;

exports[`Tests for GroupsMenuFilter Test GroupsMenuFilter for wordAlignment 1`] = `
<div
  id="groups-menu-filter"
  style={
    Object {
      "color": "var(--reverse-color)",
      "width": 238,
    }
  }
>
  <div
    id="groups-menu-filter-top"
  >
    <span
      id="groups-menu-title"
    >
      Menu
    </span>
    <Glyphicon
      bsClass="glyphicon"
      glyph="filter"
      key="filter"
    />
  </div>
  <div
    id="groups-menu-filter-options"
  >
    <div
      className="option"
    >
      <span
        className="object-checkbox"
      >
        <input
          checked={true}
          name="showInvalidated"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="object-icon"
      >
        <InvalidatedIcon
          color="#fff"
          height={16}
          width={16}
        />
      </span>
      <span
        className="object-text"
      >
        Invalidated
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showBookmarks"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-icon"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="bookmark"
        />
      </span>
      <span
        className="option-text"
      >
        Bookmarks
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showSelections"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-icon"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="ok"
        />
      </span>
      <span
        className="option-text"
      >
        Selected
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showNoSelections"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="ban-circle"
        />
      </span>
      <span
        className="option-text"
      >
        No Selection
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showVerseEdits"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="pencil"
        />
      </span>
      <span
        className="option-text"
      >
        Verse edit
      </span>
    </div>
    <div
      className="option"
    >
      <span
        className="option-checkbox"
      >
        <input
          checked={true}
          name="showComments"
          onChange={[Function]}
          type="checkbox"
        />
      </span>
      <span
        className="option-checkbox"
      >
        <Glyphicon
          bsClass="glyphicon"
          glyph="comment"
        />
      </span>
      <span
        className="option-text"
      >
        Comments
      </span>
    </div>
  </div>
</div>
`;
